aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1156,6 +1156,25 @@ std::vector<Tensor> gradient(const Tensor& self, IntArrayRef dim, int64_t edge_o
 
 // ALL REDUCE #################################################################
 
+inline bool should_use_acc_buffer(at::TensorIterator& iter) {
+  const auto ndim = iter.ndim();
+  if (!iter.device().is_cpu()) {
+    return false;
+  }
+  if (!at::isReducedFloatingType(iter.common_dtype())) {
+    return false;
+  }
+  if (ndim < 2 || iter.noutputs() != 1) {
+    return false;
+  }
+  auto out_strides = iter.strides(0);
+  for (const auto dim : c10::irange(0, 2)) {
+      if (out_strides[dim] != 0) {
+        return false;
+      }
+  }
```

Comment:
should check that we have at least three reduction dims here?

Reply:
Checking `ndim` should have the same effect.

Reply:
Not sure if I understand but you only check the first two dims here for reduction dims. Is that correct?

Reply:
Yes, only check the first two dims. For example: 
Output stride is [x, x, x, x, ..., n] with x >= 0 and  n > 0. 
If the stride of the first two dims is 0 (we can also only check whether the second dim is 0), the second dim may accumulate intermediate sums which are generated on the first dim.

Reply:
Got it.